prov/rxm/src/rxm.h

```diff
@@ -977,6 +978,36 @@ static inline int rxm_cq_write_recv_comp(struct rxm_rx_buf *rx_buf,
 				    rx_buf->pkt.hdr.tag);
 }
 
+static inline enum fi_hmem_iface rxm_iov_to_hmem_iface(const struct iovec *iov,
+						       size_t count)
+{
+	int ret;
+	enum fi_hmem_iface iface;
+
+	if (!count)
+		return FI_HMEM_SYSTEM;
+
+	ret = ofi_get_hmem_iface(iov[0].iov_base, &iface);
```

Comment:
If an app knows enough to request FI_HMEM* capability, it already has knowledge about the type of buffer it is going to allocate and use, would it not be better to require FI_MR_HMEM in that case?

Reply:
I have mentioned this. But, the use case they are looking for rxm to support is FI_HMEM set with FI_MR_HMEM and FI_MR_LOCAL unset.